plugin/src/main/scala/ch/epfl/scala/ProfilingPlugin.scala

```diff
@@ -96,10 +103,31 @@ class ProfilingPlugin(val global: Global) extends Plugin {
     }
 
     private def reportStatistics(graphsPath: AbsolutePath): Unit = {
-      val macroProfiler = implementation.macroProfiler
-      val persistedGraphData = implementation.generateGraphData(graphsPath)
+      val globalDir =
+        if (config.generateGlobalFlamegraph) {
+          val scalaDir =
+            if (ScalaSettingsOps.isScala212)
+              "scala-2.12"
+            else if (ScalaSettingsOps.isScala213)
+              "scala-2.13"
+            else
+              sys.error(s"Currently, only Scala 2.12 and 2.13 are supported, " +
+                s"but [${global.settings.source.value}] has been spotted")
+
+          val globalDir =
+            ProfileDbPath.toGraphsProfilePath(
+              config.sourceRoot.resolve(RelativePath(s"target/$scalaDir/classes"))
```

Comment:
The target directory is customizable un sbt. This doesn't work if you customize it. Maybe the plugin should receive the ScalaDir directory as parameter which opens the door to use the plugin also in other build tools like mill where the directory structure is different.

Reply:
Yeah, I thought about this one. If we agree on the approach, I will iterate on this. From my understanding, `scalac-profiling` was designed for SBT mostly. There are also a few other things related to SBT's output.

Reply:
In my projects at work I have:
```scala
// avoid invalidating compilation cache when enabling/disabling coverage
target := { if (ScoverageKeys.coverageEnabled.value) target.value / "coverage" else target.value }
```
which will break here. But we don't need to fix this problem now. It can be iterated over later :)

Reply:
@lolgab but this would also change targets for particular modules, and currently generated flamegraphs will be put into the `coverage` space, won't it?

Reply:
Yes. but your hardcoding of `target` will not work.

Reply:
So perhaps we need to generalize the task and make target configurable overall, wdyt?

Reply:
> but your hardcoding of target will not work.

btw, have you tried specifying the `source root` directory?

Reply:
You can ask for `crossTarget` and then look inside for the `classes`.
For example it can be something like: `s"-P:scalac-profiling:crossTarget=${crossTarget.value}"`.
`crossTarget` points by default to `target/scala-2.13` but it can be configured and matches what sbt uses.
If you change `target` sbt automatically changes `crossTarget`.

Reply:
But yeah, it doesn't need to be done in this PR. Just something to keep in mind for later.